main.py

```
@@ -625,6 +626,19 @@ def _on_extract_call(result):
+        # C-prime spec §5.1: cap exceeded is a server-side generation failure,
+        # not a client input failure → 500 (not 422). Must be caught BEFORE the
+        # generic ValueError handler below because SmallestRouteCapExceeded
+        # subclasses ValueError.
+        logger.error("extract: smallest_route_cap_exceeded: %s", err)
+        raise HTTPException(
+            status_code=500,
+            detail={
+                "error": "smallest_route_cap_exceeded",
+                "message": "Could not generate a valid starting map. Try again or adjust the prompt.",
+            },
+        )
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)